x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/stringstats/InternalStringStatsTests.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.analytics.stringstats;
+
+import org.elasticsearch.client.analytics.ParsedStringStats;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.Aggregation;
+import org.elasticsearch.search.aggregations.ParsedAggregation;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.InternalAggregationTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static java.util.Collections.emptyMap;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class InternalStringStatsTests extends InternalAggregationTestCase<InternalStringStats> {
+    @Override
+    protected List<NamedXContentRegistry.Entry> getNamedXContents() {
+        List<NamedXContentRegistry.Entry> result = new ArrayList<>(super.getNamedXContents());
+        result.add(new NamedXContentRegistry.Entry(Aggregation.class, new ParseField(StringStatsAggregationBuilder.NAME),
+                (p, c) -> ParsedStringStats.PARSER.parse(p, (String) c)));
+        return result;
+    }
+
+    protected InternalStringStats createTestInstance(
+            String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
+        if (randomBoolean()) {
+            return new InternalStringStats(name, 0, 0, 0, 0, emptyMap(), randomBoolean(), DocValueFormat.RAW,
+                    pipelineAggregators, metaData);
+        }
+        return new InternalStringStats(name, randomLongBetween(1, Long.MAX_VALUE),
+                randomNonNegativeLong(), between(0, Integer.MAX_VALUE), between(0, Integer.MAX_VALUE), randomCharOccurrences(),
+                randomBoolean(), DocValueFormat.RAW,
+                pipelineAggregators, metaData);
+    };
+
+    @Override
+    protected InternalStringStats mutateInstance(InternalStringStats instance) throws IOException {
+         String name = instance.getName();
+         long count = instance.getCount();
+         long totalLength = instance.getTotalLength();
+         int minLength = instance.getMinLength();
+         int maxLength = instance.getMaxLength();
+         Map<String, Long> charOccurrences = instance.getCharOccurrences();
+         boolean showDistribution = instance.getShowDistribution();
+         switch (between(0, 6)) {
+         case 0:
```

Comment:
That indentation looks a bit strange to me.

Reply:
It is the true indentation for case statements!

Reply:
It is [indeed](https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html), we just don't use it anywhere else :) that's why it looks strange.

Reply:
It is the Eclipse standard but not the intellij standard. I prefer this was but I'm not picky.